api/users/views.py

```diff
@@ -401,8 +405,48 @@ def get_default_odm_query(self):
             Q('is_registration', 'eq', True) &
             Q('contributors', 'eq', user._id)
         )
+
         permission_query = Q('is_public', 'eq', True)
         if not current_user.is_anonymous():
             permission_query = (permission_query | Q('contributors', 'eq', current_user._id))
         query = query & permission_query
         return query
+
+class UserNodeLogs(JSONAPIBaseView, generics.ListCreateAPIView, ODMFilterMixin, UserMixin):
+
+    serializer_class = NodeLogSerializer
+    pagination_class = UserNodeLogPagination
+    view_category = 'users'
+    view_name = 'node_logs'
+
+    ordering = ('-date', )
+
+    def get_default_odm_query(self):
+        user = self.get_user()
+        if user.is_anonymous():
+            auth = Auth(None)
+        else:
+            auth = Auth(user)
+        raw_nodes = Node.find(
+            Q('contributors', 'eq', user) &
+            Q('is_folder', 'ne', True) &
+            Q('is_deleted', 'ne', True)
+        )
+        nodes = [each for each in raw_nodes if each.is_public or each.can_view(auth)]
+        if not nodes:
+            raise NotFound
+        query = [Q('params.node', 'eq', node._id) for node in nodes]
+        return reduce(lambda x, y: x | y, query)
```

Comment:
Can you explain to me how this works? :) I see that it is returning your list of queries into a query group so you can then use it to query NodeLogs but I don't understand how it's doing that.